src/kademlia/dht_storage.cpp

```diff
@@ -257,7 +256,7 @@ namespace
 					++m;
 				}
 			}
-			return true;
+			return int(i->second.peers.size()) >= m_settings.max_peers;
```

Comment:
It definitely shouldn't be false, we want to withhold a write token if the torrent is at the max number of peers. But now that you've made me think about it again I've realized there's a problem, even if the torrent is at max_peers we need to send a token if the requester is already in the peer set.